x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mapping.tsx

```diff
@@ -0,0 +1,292 @@
+  {
+    type: 'owner',
+    description: OWNER_ROLE_TYPE_DESCRIPTION,
+  },
+  {
+    type: 'admin',
+    description: ADMIN_ROLE_TYPE_DESCRIPTION,
+  },
+] as AdvanceRoleType[];
+
+export const RoleMapping: React.FC<RoleMappingProps> = ({ isNew }) => {
+  const { roleId } = useParams() as { roleId: string };
+  const { myRole } = useValues(AppLogic);
+
+  const {
+    handleAccessAllEnginesChange,
+    handleAttributeSelectorChange,
+    handleAttributeValueChange,
+    handleAuthProviderChange,
+    handleDeleteMapping,
+    handleEngineSelectionChange,
+    handleRoleChange,
+    handleSaveMapping,
+    initializeRoleMapping,
+    resetState,
+  } = useActions(RoleMappingsLogic);
+
+  const {
+    accessAllEngines,
+    attributeName,
+    attributeValue,
+    attributes,
+    availableAuthProviders,
+    availableEngines,
+    dataLoading,
+    elasticsearchRoles,
+    hasAdvancedRoles,
+    multipleAuthProvidersConfig,
+    roleMapping,
+    roleType,
+    selectedEngines,
+    selectedAuthProviders,
+  } = useValues(RoleMappingsLogic);
+
+  useEffect(() => {
+    initializeRoleMapping(roleId);
+    return resetState;
+  }, []);
+
+  if (dataLoading) return <Loading />;
+
+  const SAVE_ROLE_MAPPING_LABEL = i18n.translate(
+    'xpack.enterpriseSearch.appSearch.roleMapping.saveRoleMappingButtonMessage',
+    {
+      defaultMessage: '{operation} role mapping',
+      values: { operation: isNew ? 'Save' : 'Update' },
+    }
+  );
+
+  const ACTIVE_BREADCRUMB_TEXT = i18n.translate(
+    'xpack.enterpriseSearch.appSearch.roleMapping.activeBreadcrumbText',
+    {
+      defaultMessage: '{operation} role mapping',
+      values: { operation: isNew ? 'Add' : 'Manage' },
+    }
+  );
+
+  const saveRoleMappingButton = (
+    <EuiButton onClick={handleSaveMapping} fill>
+      {SAVE_ROLE_MAPPING_LABEL}
+    </EuiButton>
+  );
+
+  const engineSelector = (engine: Engine) => (
+    <EuiCheckbox
+      key={engine.name}
+      name={engine.name}
+      id={`engine_option_${engine.name}`}
+      checked={selectedEngines.has(engine.name)}
+      onChange={(e) => {
+        handleEngineSelectionChange(engine.name, e.target.checked);
+      }}
+      label={engine.name}
+    />
+  );
+
+  const advancedRoleSelectors = (
+    <>
+      <EuiSpacer />
+      <EuiTitle size="xs">
+        <h4>{ADVANCED_ROLE_SELECTORS_TITLE}</h4>
+      </EuiTitle>
+      <EuiSpacer />
+      {advancedRoleTypes.map(({ type, description }) => (
+        <RoleSelector
+          key={type}
+          disabled={!myRole.availableRoleTypes.includes(type)}
+          roleType={roleType}
+          roleTypeOption={type}
+          description={description}
+          onChange={handleRoleChange}
+        />
+      ))}
+    </>
+  );
+
+  return (
+    <>
+      <SetPageChrome trail={[ROLE_MAPPINGS_TITLE, ACTIVE_BREADCRUMB_TEXT]} />
+      <EuiPageHeader rightSideItems={[saveRoleMappingButton]} pageTitle={ROLE_MAPPINGS_TITLE} />
+      <EuiSpacer />
+      <EuiPageContent>
```

Comment:
Davey might want us to drop the `EuiPageContent` wrapper here - it's adding an `EuiPanel` wrapper that looks odd when wrapped against other nested `EuiPanel` children. IMO just the `<EuiPageContentBody>` wrapper suffices for grouping this content together!

Reply:
https://github.com/elastic/kibana/pull/94763/commits/9b5c04df582c03cef12bf2722d7f74198927b617